pkgs/applications/science/electronics/kingstvis/default.nix

```diff
@@ -0,0 +1,72 @@
+{ stdenv
+, buildFHSEnv
+, lib
+, fetchurl
+, glib
+, xorg
+, libGL
+, zlib
+, fontconfig
+, freetype
+, xkeyboard_config
+, dbus
+}:
+
+let
+
+  pkg = stdenv.mkDerivation (rec {
+    pname = "kingstvis";
+    version = "3.6.1";
+
+    src = fetchurl {
+      url = "http://res.kingst.site/kfs/KingstVIS_v3.6.1.tar.gz";
+      sha256 = "sha256-WWLlUm7031E8v5EVobXpNo9sURh+S4aVfmhDlWjK7qA=";
+    };
+
+    dontBuild = true;
+    installPhase = ''
+      cp -R . $out
+    '';
+  });
```

Comment:
If we are just unpacking the files, we could also inline this to avoid the extra derivation

Reply:
I'm not entirely sure how to accomplish this inlining, but I replaced the "mkDerivation" with a "fetchzip"

```nix
let
  name = "kingstvis";
  version = "3.6.1";
  src = fetchzip {
    url = "http://res.kingst.site/kfs/KingstVIS_v${version}.tar.gz";
    hash = "sha256-eZJ3RZWdmNx/El3Hh5kUf44pIwdvwOEkRysYBgUkS18=";
  };
in
buildFHSEnv {
  inherit name;

  extraInstallCommands = ''
    install -Dvm644 ${src}/Driver/99-Kingst.rules \
      $out/lib/udev/rules.d/99-Kingst.rules
  '';

  runScript = "${src}/KingstVIS";
```

Is something like that what you meant?

Reply:
yes :)